### PR TITLE
TLS Verifying peer in QUIC layer is supported

### PR DESCRIPTION
**Pull request by wanghaEMQ**
* TLS Verifying peer in QUIC layer is supported.
* Fix heap overflow error when set tls ca/cert/pwd.
* Fix a warning.

### CODE REVIEW

**Review by OdyWayne — APPROVED** [review]

**Review by JaylinYu — COMMENTED** [review]
have u tested it?

File: src/supplemental/quic/quic_api.c [hunk 1]
```diff
@@ -189,8 +190,8 @@ quic_dialer_set_tls_pwd(void *arg, const void *buf, size_t sz, nni_type t)
 	int              rv;
 	char *           str;
 
-	str = nng_alloc(sz);
-	memset(str, '\0', sz);
+	str = nng_alloc(sz + 1);
```
Comment (JaylinYu) [inline src/supplemental/quic/quic_api.c:193]: no need to add offset here

**Review by JaylinYu — APPROVED** [review]
I misplaced Ca and cert?

**Comment by wanghaEMQ** [comment]
> have u tested it?

yes. Of course I did

**Comment by wanghaEMQ** [comment]
> I misplaced Ca and cert?

Yes.

**Pull request merged**